notebooks/4.visualization/Figure2_model_performance_and_eval.r

```diff
@@ -0,0 +1,303 @@
+    dataset_type == "Test" & shuffled == "Shuffled" ~ "Test + Shuffled",
+    dataset_type == "Train" & shuffled == "Not Shuffled" ~ "Train + Not Shuffled",
+    dataset_type == "Train" & shuffled == "Shuffled" ~ "Train + Shuffled",
+  )) %>%
+  mutate(is_train_test = if_else(dataset_type %in% c("test", "train"), "test_train", "plate_holdout"))
+
+# plot data
+width <- 10
+height <- 10
+options(repr.plot.width = width, repr.plot.height = height)
+
+pr_curve_plot_train_test <- ggplot(filtered_pr_df, aes(x = recall, y = precision)) +
+    geom_line(aes(color = dataset_type, linetype = shuffled)) +
+    facet_wrap(~injury_type) +
+    xlab("Recall") +
+    ylab("Precision") +
+    theme_bw() +
+    theme(
+        legend.spacing.y = unit(0.1, "cm"),
+        legend.box.spacing = unit(0.1, "cm"),
+        legend.key.size = unit(0.7, "lines"),
+        legend.key.width = unit(1, "lines"),
+        axis.text.x = element_text(angle = 90, size = 14),
+        axis.text.y = element_text(size = 14),
+        strip.text = element_text(size = 13),
+        axis.title = element_text(size = 14),
+        legend.title = element_text(size = 15),
+        legend.text = element_text(size = 14)) +
+    labs(color = "Data Split", linetype = "Shuffled")
+
+ggsave("figures/only_test_train_pr_curve.png", width = width, height = height, dpi=600)
+
+pr_curve_plot_train_test
+
+# image size
+img_height <- 10
+img_width <- 15
+
+options(repr.plot.width = img_width, repr.plot.height = img_height)
+probabilities_ridge_plot <- (
+    ggplot(proba_df, aes(x = proba, y = pred_injury, fill = shuffled_model)) +
+    geom_density_ridges() +
+    facet_wrap(~injury_type) +
+    scale_x_continuous(breaks = seq(0, 1, 0.5)) +
+    theme_bw() +
+
+    # no legend
+    theme(plot.title = element_text(size = 20, hjust = 0.5)) +
+
+    scale_fill_manual(values = c(
+    "Shuffled" = brewer.pal(2, "Dark2")[2],
+    "Not Shuffled" = "#1E93FC"
+    )) +
+
+    theme(
+        axis.title.x = element_blank(),
+        legend.spacing.y = unit(0.1, "cm"),
+        legend.box.spacing = unit(0.1, "cm"),
+        legend.key.size = unit(0.7, "lines"),
+        legend.key.width = unit(1, "lines"),
+        strip.text = element_text(size = 12),
+        axis.text.y = element_text(size = 10),
+        axis.title = element_text(size = 14),
+        axis.title.x.bottom = element_text(margin = margin(t = 15)),
+        axis.title.y.left = element_text(margin = margin(t = 15)),
+        legend.title = element_text(size = 14),
+        legend.text = element_text(size = 12)
+        )) +
+
+    labs(y = "Injury Types", x = "Probability", fill = "Model Type")
+
+
+ggsave(filename = "figures/JUMP_cell_injury_facet_proba.png", height = img_height, width = img_width, dpi=600)
+probabilities_ridge_plot
+
+# plot dims
+height = 20
+width = 25
+
+options(repr.plot.width=width, repr.plot.height=height, units = "cm", dpi = 600)
+
+# Combine plots using patchwork
+top_plot<- (
+  pr_curve_plot_train_test |
```

Comment:
```suggestion
  wrap_elements(full = pr_curve_plot_train_test) |
```

Try this out and see if it helps with white space.

Reply:
I did some part but still didn't solve the weight space produced in figure D